spring-cloud-commons/src/main/java/org/springframework/cloud/client/serviceregistry/AbstractAutoServiceRegistration.java

```diff
@@ -130,16 +135,45 @@ public void start() {
 		// because of containerPortInitializer below
 		if (!this.running.get()) {
 			this.context.publishEvent(new InstancePreRegisteredEvent(this, getRegistration()));
+			List<RegistrationLifecycle> registrationLifecycleList = getRegistrationLifecycles();
+			registrationLifecycleList.forEach(
+					registrationLifecycle -> registrationLifecycle.postProcessBeforeStartRegister(getRegistration()));
 			register();
+			registrationLifecycleList.forEach(
+					registrationLifecycle -> registrationLifecycle.postProcessAfterStartRegister(getRegistration()));
 			if (shouldRegisterManagement()) {
+				List<RegistrationManagementLifecycle> registrationManagementLifecycles = getRegistrationManagementLifecycles();
+				registrationManagementLifecycles
+						.forEach(registrationManagementLifecycle -> registrationManagementLifecycle
+								.postProcessBeforeStartRegisterManagement(getManagementRegistration()));
 				registerManagement();
+				registrationManagementLifecycles
+						.forEach(registrationManagementLifecycle -> registrationManagementLifecycle
+								.postProcessAfterStartRegisterManagement(getManagementRegistration()));
+
 			}
 			this.context.publishEvent(new InstanceRegisteredEvent<>(this, getConfiguration()));
 			this.running.compareAndSet(false, true);
 		}
 
 	}
 
+	private List<RegistrationManagementLifecycle> getRegistrationManagementLifecycles() {
```

Comment:
I would prefer these be passed in via the constructor rather be retrieved from the context in this case.

Reply:
My intention is to get it directly from the container, not from a member variable.  However,I'm going to modify this part of the code